### ISSUE TITLE
Fatal Exception: DateUtils

### ISSUE THREAD

**Issue by AmandaRiu** [woocommerce/woocommerce-android#392]
I was finally able to reproduce the `DateUtils` Fatal Exception by switching tabs after reloading the dashboard several times:

```
java.lang.IllegalArgumentException: Date string argument is not of format YYYY-'W'WW: 2018-08-23
```

It appears to be a timing issue which can be fixed several ways, but I think the best thing would probably be to make this method more forgiving. It should parse year/month from the provided string regardless if the day is also present. So it should just parse out month by index instead of using `.last()`

```
@Throws(IllegalArgumentException::class)
fun getShortMonthString(iso8601Month: String): String {
    val month = iso8601Month.split("-").last()
    return try {
        shortMonths[month.toInt() - 1]
    } catch (e: Exception) {
        throw IllegalArgumentException("Date string argument is not of format YYYY-MM: $iso8601Month")
    }
}
```

**Comment by jkmassel** [comment]
This has popped back up in the `1.6` beta.

**Source:** `5cb849aef8b88c29637fdc2f-fabric`

**Comment by anitaa1990** [comment]
@AmandaRiu @AmandaRiu 
This seems to be happening when user is trying to view shipment tracking details. 
```
Fatal Exception: java.lang.IllegalArgumentException
Date string argument is not of format YYYY-MM-DD: 2019-04-18
com.woocommerce.android.util.DateUtils.getFullDateString (DateUtils.java:119)
```
Some points I've gathered:
* This method is no longer being used in the app and is replaced with `getLocalizedLongDateString(` from 1.7 onwards.
* This issue was faced by only 1 user. I’ve SSP’ed into the user’s account on 1.6 and can no longer see the shipment tracking information for the user (because they disabled/deleted the Shipment tracking plugin).
* I tried to reproduce the issue on my test site but wasn’t successful in that.

At this point, I think we can wait till we get any similar reports, and if we might be able to find out more about this crash. 

Having said that, I am very much open to any other suggestions regarding how to proceed next.
